modules/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,98 @@
+package kubernetes
+
+import (
+	"context"
+	_ "embed"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/odpf/entropy/core/module"
+	"github.com/odpf/entropy/core/resource"
+	"github.com/odpf/entropy/pkg/errors"
+)
+
+//go:embed config_schema.json
+var configSchema string
+
+var KubeModule = module.Descriptor{
+	Kind: "kubernetes",
+	Actions: []module.ActionDesc{
+		{
+			Name:        module.CreateAction,
+			ParamSchema: configSchema,
+		},
+		{
+			Name:        module.UpdateAction,
+			ParamSchema: configSchema,
+		},
+	},
+	Module: &kubeModule{},
+}
+
+type kubeModule struct{}
+
+func (k *kubeModule) Plan(_ context.Context, spec module.Spec, act module.ActionRequest) (*resource.Resource, error) {
+	res := spec.Resource
+
+	cfg := &rest.Config{
+		Host: act.Params["host"].(string),
+		TLSClientConfig: rest.TLSClientConfig{
+			CAData:   []byte(act.Params["ca_cert"].(string)),
+			KeyData:  []byte(act.Params["client_key"].(string)),
+			CertData: []byte(act.Params["client_cert"].(string)),
+		},
+		Timeout: time.Duration(act.Params["client_timeout"].(float64)) * time.Millisecond,
```

Comment:
it is getting converted into `int64` when we do the `time.Duration()` conversion..

But the `.(float64)` is only for converting `interface{}` to the *actual type* of the underlying value. Since JSON supports only `number` type which is int+float, Go json encoding always uses `float64` for numeric values.